kernel/src/net/socket/netlink/kobject_uevent/message/syn_uevent.rs

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! The synthetic uevent.
+//!
+//! The event is triggered when someone writes some content to `/sys/.../uevent` file.
+//! It differs from the event triggers by devices.
+//!
+//! Reference: <https://elixir.bootlin.com/linux/v6.0.9/source/Documentation/ABI/testing/sysfs-uevent>.
+
+use alloc::format;
+use core::str::FromStr;
+
+use super::uevent::KobjectAction;
+use crate::prelude::*;
+
+/// The synthetic uevent.
+pub struct SyntheticUevent {
+    pub(super) action: KobjectAction,
+    pub(super) uuid: Option<Uuid>,
+    pub(super) envs: BTreeMap<String, String>,
+}
+
+impl FromStr for SyntheticUevent {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        let split = s.split(" ").collect::<Vec<_>>();
```

Comment:
I guess we don't really need to collect this into a vector. We can just iterate `s.split(" ")` directly to avoid unnecessary heap allocations.